src/script/cpp_api/s_server.cpp

```diff
@@ -121,7 +126,8 @@ bool ScriptApiServer::setPassword(const std::string &playername,
 		throw LuaError("Authentication handler missing set_password");
 	lua_pushstring(L, playername.c_str());
 	lua_pushstring(L, password.c_str());
-	PCALL_RES(lua_pcall(L, 2, 1, m_errorhandler));
+	PCALL_RES(lua_pcall(L, 2, 1, error_handler));
+	lua_remove(L, -2); // Remove error handler
```

Comment:
I guess you use `lua_remove` here because there are still elements on the stack. But why not use `error_handler` instead of -2 as index on the stack?